torchgeo/datasets/mmearth.py

```diff
@@ -0,0 +1,505 @@
+        'dynamic_world': 0,
+        'esa_worldcover': 255,
+        'lat': float('-inf'),
+        'lon': float('-inf'),
+        'month': float('-inf'),
+        'era5': float('inf'),
+        'biome': 255,
+        'eco_region': 65535,
+    }
+
+    norm_modes: ClassVar[list[str]] = ['z-score', 'min-max']
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        ds_version: str = 'MMEarth',
+        modalities: Sequence[str] = all_modalities,
+        modality_bands: dict[str, list[str]] | None = None,
+        split: str = 'train',
+        normalization_mode: str = 'z-score',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+    ) -> None:
+        """Initialize the MM-Earth dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            ds_version: one of "MMEarth", "MMEarth64", or "MMEarth100k"
+            modalities: list of modalities to load
+            modality_bands: dictionary of modality bands, see `all_modality_bands`
+            split: one of "train", "val", or "test"
+            normalization_mode: one of "z-score" or "min-max"
+            transforms: a function/transform that takes input sample dictionary
+                and returns a transformed version
+
+        """
+        lazy_import('h5py')
+
+        assert (
+            normalization_mode in self.norm_modes
+        ), f'Invalid normalization mode: {normalization_mode}, please choose from {self.norm_modes}'
+        assert (
+            ds_version in self.ds_versions
+        ), f'Invalid dataset version: {ds_version}, please choose from {self.ds_versions}'
+        assert (
+            split in self.splits
+        ), f'Invalid split: {split}, please choose from {self.splits}'
+
+        self._validate_modalities(modalities)
+        if modality_bands is None:
+            modality_bands = {
+                modality: self.all_modality_bands[modality] for modality in modalities
+            }
+        self._validate_modality_bands(modality_bands)
+
+        self.modalities = modalities
+        self.modality_bands = modality_bands
+
+        self.root = root
+        self.ds_version = ds_version
+        self.normalization_mode = normalization_mode
+        self.split = split
+        self.transforms = transforms
+
+        self.dataset_filename = f'{self.filenames[ds_version]}.h5'
+        self.band_stats_filename = f'{self.filenames[ds_version]}_band_stats.json'
+        self.splits_filename = f'{self.filenames[ds_version]}_splits.json'
+        self.tile_info_filename = f'{self.filenames[ds_version]}_tile_info.json'
+
+        self._verify()
+
+        self.indices = self._load_indices()
+        self.band_stats = self._load_normalization_stats()
+        self.tile_info = self._load_tile_info()
+
+    def _verify(self) -> None:
+        """Verify the dataset.
+
+        Raises:
+            AssertionError: if dataset files are not found
```

Comment:
Not true. I usually move the errors to the `__init__` docstring since `_verify` won't be included in the docs.